components/board/pinwrappers/analogs.go

```diff
@@ -104,6 +110,14 @@ func (as *AnalogSmoother) Start() {
 	as.workers = utils.NewStoppableWorkers(func(ctx context.Context) {
 		consecutiveErrors := 0
 		var lastError error
+
+		// Store the full analog reader value
+		analogVal, err := as.Raw.Read(ctx, nil)
+		as.lastError.Store(&errValue{err != nil, err})
+		as.mu.Lock()
+		as.analogVal = analogVal
+		as.mu.Unlock()
```

Comment:
This makes it look like the point of the mutex is to protect `analogVal`. If that's the case, we should also lock the mutex before line 65. I wonder if lines 70-71 (locking and deferring unlocking the mutex) should just be moved to the top of the function for simplicity. 

Another option is to change `Read` to never return `as.analogVal` itself, but always to create a new `board.AnalogValue` struct instead and copy data into that. Then, `as.analogVal` is constant after this line, and we don't need the mutex at all.

Reply:
did the second option and removed the mutex